axlearn/common/trainer.py

```diff
@@ -271,8 +279,11 @@ def __init__(
                     "xsc_check_policy was set for non-TPU XLA backend. Running without XSC."
                 )
             else:
+                if cfg.cache_python_train_step is True:
+                    raise ValueError("cache_python_train_step cannot be True when xsc is enabled.")
```

Comment:
Actually `cache_python_train_step` is still useful even with XSC, since the non-XSC steps will be cached and only the XSC step needs to be recompiled. The XSC step does not run often, so maybe this is OK?